core/local/analysis.js

```diff
@@ -170,12 +120,82 @@ function analyseEvents (events /*: LocalEvent[] */, pendingChanges /*: LocalChan
   }
 
   log.trace('Flatten changes map...')
-  for (let a of changesByInode.values()) changes.push(a)
+  const changes /*: LocalChange[] */ = changesFound.flush()
 
   stopMeasure()
   return changes
 }
 
+function analyseEvent (e /*: LocalEvent */, previousChanges /*: LocalChangeMap */) /*: ?LocalChange|true */ {
```

Comment:
Hard to determine what this function should result given some parameters.

Reply:
I'm not completely clear whether the return type should be something like `?NewChange|PreviousChangeUpdated` or just `?LocalChange` (in which case maybe we would be able to replace the previous change in the caller). So I didn't want to invest too much work yet.